lib/pyld/jsonld.py

```diff
@@ -4504,7 +4504,7 @@ def _value_match(self, pattern, value):
 
         if not v2 and not t2 and not l2:
             return True
-        if not (v1 in v2 or _is_empty_object(v2[0])):
+        if not (v1 in v2 or not v2 or _is_empty_object(v2[0])):
```

Comment:
This should probably be:

```suggestion
        if not (v1 in v2 or (v2 and _is_empty_object(v2[0]))):
```

Ideally tests would confirm.